### DISCUSSION TITLE
How difficult is it to add non compatible devices byself

### DISCUSSION THREAD

**Discussion by chris-010** [hrvach/deskhop#119] [General]
Hi, i purchased a very cheap keyboard with touchpad combination from china ([link](https://www.aliexpress.com/item/1005006117751578.html)), just to get a small and mobile development kit, not for daily use. It's only partially supported by deskhop, this means keyboard is working, but touchpad not. I don't want to waste development time from hrvach to get crap like this working. I just want to ask about a guideline to get non compatible devices working, when implementation will be done byself.
hrvach, do you have any hints please?

**Comment by hrvach** [comment]
Hey, please share USB descriptors - the more data points the better. Ideally everything should work on the first try, but this is not always the case - even the Linux kernel has a folder full of HID exceptions so how do the rest of us even have a fighting chance :)

**Comment by chris-010** [comment]
root@ubuntu:/home/chris/hidrd# sudo usbhid-dump

001:004:001:DESCRIPTOR         1721746939.419783
 05 01 09 02 A1 01 85 02 09 01 A1 00 05 09 19 01
 29 05 15 00 25 01 95 05 75 01 81 02 95 01 75 03
 81 01 05 01 09 30 09 31 09 38 15 80 25 7F 75 08
 95 03 81 06 05 0C 0A 38 02 95 01 81 06 C0 C0 05
 0C 09 01 A1 01 85 01 15 00 26 3C 04 19 00 2A 3C
 04 95 01 75 10 81 00 C0 05 01 09 80 A1 01 85 03
 05 01 19 81 29 83 15 00 25 01 95 03 75 01 81 06
 95 01 75 05 81 01 C0
 
001:004:000:DESCRIPTOR         1721746939.430126
 05 01 09 06 A1 01 05 07 19 E0 29 E7 15 00 25 01
 75 01 95 08 81 02 95 01 75 08 81 03 95 05 75 01
 05 08 19 01 29 05 91 02 95 01 75 03 91 03 95 06
 75 08 15 00 26 FF 00 05 07 19 00 2A FF 00 81 00
 C0
 
001:002:000:DESCRIPTOR         1721746939.436846
 05 01 09 02 A1 01 09 01 A1 00 05 09 19 01 29 05
 15 00 25 01 95 05 75 01 81 02 95 01 75 03 81 03
 05 01 09 38 15 81 25 7F 75 08 95 01 81 06 05 0C
 0A 38 02 95 01 81 06 75 08 95 01 81 03 05 01 09
 30 09 31 15 00 26 FF 7F 35 00 46 FF 7F 75 10 95
 02 81 02 C0 C0
